admins/README.md

```diff
@@ -0,0 +1,129 @@
+
+#### Zuul Launcher
+The zuul launcher is a gearman client that uses ansible to run the specified test job on the specified type of test slave. It populates several environment variables in the test environment so that the test can know what git ref to fetch and what url to fetch it from. The zuul-cloner tool can be used to checkout the correct version of multiple projects based on these environment variables.
+
+We currently run a single zuul launcher on the zuul host, but this component is horizontally-scalable and new instances should be run on separate small hosts. Once a zuul launcher connects to the zuul server via gearman, it will immediately begin handling requests.
+
+#### Zuul Web App
+The zuul web app is a very basic web server that serves /status.json, a json object that describes the current state of all pipelines. You can use the script zuul-changes.py from zuul's tools/ directory to translate the json object into shell commands that will restore the current state of the zuul pipelines -- useful for restarting the zuul server, which loses pipeline state.
+
+### Nodepool
+Nodepool manages a dynamic pool of Openstack VMs. It can be configured with multiple image types, label types, and VM providers. The images are built using diskimage-builder. A label may contain more than one image (refered to as subnodes), and a separate resource pool is maintained for each label. A provider is an Openstack cloud capable of booting the specified label(s). For more information, refer to [nodepool's documentation](http://docs.openstack.org/infra/nodepool/).
+
+#### Nodepool Server
+By default the nodepool server handles all aspects of managing the resources; but the builder, launcher, and deleter can be run in separate processes rather than separate threads for scaling and performance. It
+
+#### Nodepool Builder
+The builder uses diskimage-builder to create images and upload them to glance.
+
+#### Nodepool Launcher
+The launcher boots VMs, maintaining a minimum count of each label in addition to bursting for demand. The demand is determined by connecting to zuul via gearman.
+
+#### Nodepool Deleter
+The deleter deletes VMs after they have finished running their assigned job. Nodepool receives job-completion messages via zeromq -- a relic left over from using Jenkins with a zeromq plugin. Not all jobs send the zeromq event to trigger node deletion, and zuul's default is to not send the message, but we have enabled it globally in the parametr-function python script.
+
+### Logs Host
+A web server serving log files as static content. All jobs should be configured to use a publisher to scp test output to this host. After test completion zuul will set the GitHub status url to the parent directory for all jobs for the test run, and on test failure zuul will also comment with a link to the log path for the failing job(s).
```

Comment:
Add a note that these are for public consumption, that the web server is open to the public.